core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -1145,9 +1141,9 @@ public final boolean hasNext() {
         }
 
         private void initAfterRebalance() {
-            //TODO - refactor to use ByeBuffer without deserializing.
+            //TODO - refactor to use OakReadBuffer without deserializing.
```

Comment:
I don't think the same. I think this comment is true and should be added to our future-work list.
It might require adding a few internal methods to support comparing two sterilized keys, but it might improve our performance in some workloads.

Reply:
"sterilized keys" is it something new due to COVID-19? :) :) 

So as a future work, do you suggest to keep hi/lo boundaries in serialized way from the start?

Reply:
Not necessarily. This should work fine with either serialized or non-serialized keys.
But obviously we should put more thought into this.

Reply:
OK, I do not see what can be done there, but I put it on the future-TODO list. So we will discuss it later.